src/common/flux/persistent-store.ts

```diff
@@ -12,13 +12,16 @@ export abstract class PersistentStore<TState> extends BaseStoreImpl<TState> {
         protected readonly idbInstance: IndexedDBAPI,
         protected readonly indexedDBDataKey: string,
         protected readonly logger: Logger,
-        private initializeWithStoreData = false,
+        private persistStoreData = false,
     ) {
         super(storeName);
     }
 
     protected async persistData(storeData: any): Promise<boolean> {
-        return await this.idbInstance.setItem(this.indexedDBDataKey, storeData);
+        if (this.persistStoreData) {
+            return await this.idbInstance.setItem(this.indexedDBDataKey, storeData);
+        }
+        return Promise.resolve(true);
```

Comment:
nit: it's preferable to just `return true;` here and let the async machinery do the `Promise`-wrapping - it results in more readable stack traces if the `Promise` ends up involved in an `Error` path.